run.ps1

```diff
@@ -0,0 +1,34 @@
+# set the base tools directory
+$toolsLocalPath = Join-Path (Get-Location) "Tools"
+$runExe = Join-Path $toolsLocalPath "Microsoft.DotNet.BuildTools.Run\run.exe"
+
+# if the tool doesn't exist then download and run the boot-strapper script
+if ((Test-Path $runExe) -eq 0)
+{
+    if ((Test-Path $toolsLocalPath) -eq 0)
+    {
+        mkdir $toolsLocalPath | Out-Null
+    }
+
+    # download boot-strapper script
+    $bootStrapperPath = Join-Path $toolsLocalPath "bootstrap.ps1"
+    Invoke-WebRequest "https://raw.githubusercontent.com/dotnet/buildtools/master/bootstrap/bootstrap.ps1" -OutFile $bootStrapperPath
+
+    # now execute it
+    Invoke-Expression "$bootStrapperPath (Get-Location) $toolsLocalPath" | Out-File (Join-Path (Get-Location) "bootstrap.log")
+    if ($LastExitCode -ne 0)
+    {
+        Write-Output "Boot-strapping failed with exit code $LastExitCode"
+        exit $LastExitCode
+    }
+
+    # copy nuget.exe into the packages directory (this is unique to the build tools repo)
+    $packagesPath = Join-Path (Get-Location) "packages"
+    Get-ChildItem (Join-Path $packagesPath "NuGet.CommandLine") -filter "nuget.exe" -recurse | foreach { copy $_.FullName $packagesPath }
+}
+
+# execute the tool using the dotnet.exe host
+$dotNetExe = Join-Path $toolsLocalPath "dotnetcli\shared\Microsoft.NETCore.App\version\dotnet.exe"
+$process = Start-Process -NoNewWindow -FilePath $dotNetExe -ArgumentList "$runExe $args" -PassThru
```

Comment:
How can I call it directly with the path being in a string?
If I used the "-Wait" option then it would block on all child processes.  Doing it this way appears to return a process object for just the process that was created (the docs are unclear on the exact behavior so this is just empirical evidence).